### PR TITLE
texlab: 4.2.0 → 4.2.1

### PR DESCRIPTION
**Pull request by kira-bruneau**
###### Description of changes

Update to the latest version: https://github.com/latex-lsp/texlab/releases/tag/v4.2.1

The man page is no longer included in the source, and is now generated with `help2man`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kira-bruneau** [comment]
Result of `nixpkgs-review pr 186010` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>texlab</li>
  </ul>
</details>

**Comment by kira-bruneau** [comment]
Result of `nixpkgs-review pr 186010` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>texlab</li>
  </ul>
</details>

File: pkgs/development/tools/misc/texlab/default.nix [hunk 1]
```diff
@@ -2,41 +2,52 @@
...
   postInstall = ''
-    installManPage texlab.1
-
     # Remove generated dylib of human_name dependency. TexLab statically
     # links to the generated rlib and doesn't reference the dylib. I
     # couldn't find any way to prevent building this by passing cargo flags.
     # See https://github.com/djudd/human-name/blob/master/Cargo.toml#L43
     rm "$out/lib/libhuman_name${stdenv.hostPlatform.extensions.sharedLibrary}"
     rmdir "$out/lib"
+  '' + lib.optionalString (!isCross) ''
```
Comment (doronbehar) [inline pkgs/development/tools/misc/texlab/default.nix:46]:
```suggestion
  ''
  # When we cross compile we cannot run the output executable to generate
  # the man page.
  + lib.optionalString (!isCross) ''
```

**Review by doronbehar — COMMENTED** [review]
It took me a moment to realise this :sweat_smile: .

**Comment by doronbehar** [comment]
Other then that, everything looks good.

File: pkgs/development/tools/misc/texlab/default.nix (hunk 1 above)
Comment (kira-bruneau) [reply pkgs/development/tools/misc/texlab/default.nix:46]: Oh thanks that's a good point! I had to do this for another package, so I forgot to add a comment here :sweat_smile:

**Comment by doronbehar** [comment]
Backport: https://github.com/NixOS/nixpkgs/pull/186125

**Pull request merged**